### PR TITLE
node: refactor handler <-> channel init boundary

### PR DESCRIPTION
**Pull request by jcorbin**

### CODE REVIEW

File: node/index.js [hunk 1]
```diff
@@ -429,15 +429,29 @@ function TChannelConnection(channel, socket, direction, remoteAddr) {
 
     if (direction === 'out') {
         self.handler.sendInitRequest();
-        self.handler.once('identify.out', function onOutIdentified(hostPort) {
+        self.handler.once('init.response', function onOutIdentified(headers) {
+            /* jshint camelcase:false */
+            var hostPort = headers.host_port;
```
Comment (Raynos) [inline node/index.js]: this is adding v2 coupling to index.js

As long as we are ok with that, then thats fine.

**Comment by Raynos** [comment]
lgtm.

**Comment by jcorbin** [comment]
refactored for less coupling:
- snake_case -> camelCase now contained in v2/handler
- passes out limited object rather than all headers

**Comment by Raynos** [comment]
lgtm.

**Comment by kriskowal** [comment]
woot

**Pull request merged**